python/tank/bootstrap/configuration_writer.py

```diff
@@ -28,6 +28,50 @@
 log = LogManager.get_logger(__name__)
 
 
+class RenameGuard(object):
```

Comment:
I'm reluctant to put that in `filesystem.py`. I understand the appeal of moving it there, but I feel this is not something that should be made public. (`sgtk.util.filesystem` is part of the official API). I feel that at the moment that this interface is half-baked and if we were to provide something like this, it should probably be an IOGuard and also deal with file/folder creation.